packages/protocol/test/MaxGasTest.t.sol

```diff
@@ -0,0 +1,269 @@
+
+        uint256 _mixHash;
+        unchecked {
+            _mixHash = block.prevrandao * variables.numBlocks;
+        }
+
+        meta.id = variables.numBlocks;
+        meta.timestamp = uint64(block.timestamp);
+        meta.l1Height = uint64(block.number - 1);
+        meta.l1Hash = blockhash(block.number - 1);
+        meta.mixHash = bytes32(_mixHash);
+        meta.txListHash = keccak256(txList);
+        meta.txListByteStart = 0;
+        meta.txListByteEnd = txListSize;
+        meta.gasLimit = gasLimit;
+        meta.beneficiary = proposer;
+        meta.treasury = L2Treasury;
+
+        if(measureConsumedGas) {
+            uint256 gasRemaining = gasleft();
+            vm.prank(proposer, proposer);
+            meta = L1.proposeBlock(abi.encode(input), txList);
+            uint256 gasRemainingAfter10Verification = gasleft();
+            console2.log("Gas used by proposeBlock() with 10 verifications:",gasRemaining - gasRemainingAfter10Verification );
+        }
+        else{
+            vm.prank(proposer, proposer);
+            meta = L1.proposeBlock(abi.encode(input), txList);
+        }
+    }
+
+
+    function proveBlockInternal(
+        address msgSender,
+        address prover,
+        TaikoData.BlockMetadata memory meta,
+        bytes32 parentHashSingle,
+        uint32 parentGasUsedSingle,
+        uint32 gasUsedSingle,
+        bytes32 blockHash,
+        bytes32 signalRoot,
+        bool measureConsumedGas
+    )
+        internal
+    {
+        TaikoData.BlockEvidence memory evidence = TaikoData.BlockEvidence({
+            metaHash: LibUtils.hashMetadata(meta),
+            parentHash: parentHashSingle,
+            blockHash: blockHash,
+            signalRoot: signalRoot,
+            graffiti: 0x0,
+            prover: prover,
+            parentGasUsed: parentGasUsedSingle,
+            gasUsed: gasUsedSingle,
+            proofs: new bytes(102)
+        });
+
+        bytes32 instance = getInstance(conf, evidence);
+        uint16 verifierId = 100;
+
+        evidence.proofs = bytes.concat(
+            bytes2(verifierId),
+            bytes16(0),
+            bytes16(instance),
+            bytes16(0),
+            bytes16(uint128(uint256(instance))),
+            new bytes(100)
+        );
+
+
+        if(measureConsumedGas) {
+            uint256 gasRemaining = gasleft();
+            vm.prank(msgSender, msgSender);
+            L1.proveBlock(meta.id, abi.encode(evidence));
+            uint256 gasRemainingAfter10Verification = gasleft();
+            console2.log("Gas used by proveBlock() with 10 verifications:",gasRemaining - gasRemainingAfter10Verification );
+        }
+        else{
+            vm.prank(msgSender, msgSender);
+            L1.proveBlock(meta.id, abi.encode(evidence));
+        }
+    }
+
+    function test_gasConsumption_with10verifications_in_a_row_with_prove_and_propose() external {
+
+        bytes memory txList2 = NonEmptyBytes121K.getNonEmpty120KBytes();
```

Comment:
Not possible to just generate a list of non-zero bytes on the fly? Or is the data inside the txList important somehow for the test to succeed?

Reply:
Sure but why not simply generate a list of 120k x 0x01 in code vs having a very long constant value that's pasted in the test code?